examples/pytorch/ogb/directional_GSN/preprocessing.py

```diff
@@ -0,0 +1,285 @@
+            # Then, increase the count of the matched edge w.r.t. the corresponding orbit
+            # Repeat for the reverse edge (the one with the opposite direction)
+            
+            edge_orbit = subgraph_dict['orbit_membership'][i]
+            mapped_edge = tuple([mapping[edge[0]], mapping[edge[1]]])
+            counts[edge_dict[mapped_edge], edge_orbit] += 1
+            
+    counts = counts/subgraph_dict['aut_count']
+    
+    counts = torch.tensor(counts)
+    
+    return counts
+
+def prepare_dataset(path, name, **subgraph_params):
+    
+    k = 8
+
+    data_folder = os.path.join(path, 'processed')
+    if not os.path.exists(data_folder):
+        os.makedirs(data_folder)
+    
+    data_file = os.path.join(data_folder, 'cycle_graph_induced_{}.pt'.format(k))
+        
+    graphs_dgl, orbit_partition_sizes, split_idx = generate_dataset(path, name, **subgraph_params)
+    if data_file is not None:
+        print("Saving dataset to {}".format(data_file))
+        torch.save((graphs_dgl, orbit_partition_sizes, split_idx), data_file)
+
+    return graphs_dgl, split_idx
+
+def generate_dataset(path, name):
+
+    ### compute the orbits of earch substructure in the list, as well as the vertex automorphism count
+    subgraph_dicts = []
+    orbit_partition_sizes = []
+
+    edge_lists = []
+    for k in range(3, 8 + 1):
+        graphs_nx = getattr(nx, 'cycle_graph')(k)
+        if isinstance(graphs_nx, list) or isinstance(graphs_nx, types.GeneratorType):
+            edge_lists += [list(graph_nx.edges) for graph_nx in graphs_nx]
+        else:
+            edge_lists.append(list(graphs_nx.edges))
+
+    for edge_list in edge_lists:
+        subgraph, orbit_partition, orbit_membership, aut_count = induced_edge_automorphism_orbits(edge_list=edge_list)
+        subgraph_dicts.append({'subgraph':subgraph, 'orbit_partition': orbit_partition, 
+                               'orbit_membership': orbit_membership, 'aut_count': aut_count})
+        orbit_partition_sizes.append(len(orbit_partition))
+        
+    ### load and preprocess dataset
+    dataset = DglGraphPropPredDataset(name=name, root=path)
+    split_idx = dataset.get_idx_split()
+        
+    # computation of subgraph isomoprhisms & creation of data structure
+    graphs_dgl = list()
+    for i, datapoint in tqdm(enumerate(dataset)):
+        g, label = datapoint
+        g = _prepare(g, subgraph_dicts)
+        graphs_dgl.append((g, label))
+
+    return graphs_dgl, orbit_partition_sizes, split_idx
+        
+def _prepare(g, subgraph_dicts):
+
+    edge_index = torch.stack(g.edges())
+    
+    identifiers = None
+    for subgraph_dict in subgraph_dicts:
+        counts = subgraph_isomorphism_edge_counts(edge_index, subgraph_dict)
+        identifiers = counts if identifiers is None else torch.cat((identifiers, counts),1) 
+
+    g.edata['subgraph_counts'] = identifiers.long()
+    
+    return g
+
+class pcbaDGL(torch.utils.data.Dataset):
```

Comment:
We follow the CamelCase for class naming convention, i.e., change `pcbaDGL` to `PCBADGL`.